src/GovUk.Education.ExploreEducationStatistics.Common/Services/PostgreSqlRepository.cs

```diff
@@ -0,0 +1,99 @@
+#nullable enable
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Requests;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using Npgsql;
+using NpgsqlTypes;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Services;
+
+public class PostgreSqlRepository : IPostgreSqlRepository
+{
+    public async Task<TResponse?> GetJsonbFromPath<TDbContext, TRowId, TResponse>(
+        TDbContext context,
+        JsonbPathRequest<TRowId> request,
+        CancellationToken cancellationToken = default)
+        where TDbContext : DbContext
+        where TResponse : class
+    {
+        var jsonPathParam = new NpgsqlParameter("jsonPath", request.PathSegments);
+        var rowIdParam = new NpgsqlParameter("rowId", request.RowId);
+
+        var response = await context
+            .Set<JsonFragment>()
+            .FromSqlRaw(
+                sql: $"""
+                      SELECT "{request.JsonbColumnName}"#>@jsonPath "JsonString" FROM "{request.TableName}"
```

Comment:
Would it be worth doing `"{nameof(JsonFragment.JsonString)}"` in place of `"JsonString"`?

Reply:
Yup absolutely, good suggestion ta